controllers/storageautoscaler/scraper.go

```diff
@@ -107,9 +112,11 @@ func filterObjectsForScaling(storageAutoScalerList *ocsv1.StorageAutoScalerList,
 			err := fmt.Errorf("osd usage is nil for device class, device class: %s", storageAutoScaler.Spec.DeviceClass)
 			log.Error(err, "device class not found in sync map")
 		} else {
-			deviceClassUsage := usage.(float64)
-			if (deviceClassUsage * 100) >= float64(deviceClassThreshold) {
-				objectList = append(objectList, types.NamespacedName{Namespace: storageAutoScaler.Namespace, Name: storageAutoScaler.Name})
+			deviceClassUsage := usage.(OsdUsage)
+			if deviceClassUsage.TotalUsage*100 >= 50 {
```

Comment:
Let's make this more dependent on the configured threshold. After some quick calculations I would suggest `deviceClassThreshold - 10%`. By default, this would be a 60% threshold, which gives some room for OSDs to have different filled ratios, but also not too low of a threshold. If the threshold is too low, it may not work as well when we are horizontally scaling and adding 3 OSDs (instead of doubling OSD sizes in vertical expansion). Proportionally, we won't be adding as much per horizontal expansion as the cluster gets bigger.

Reply:
Correct I did some calculations too,

if we are scaling horizontally after 8tb osds,

Total capacity we increase like, 
24, 48, 72, 96......, 216, 240, 268...

248 ---> 268 => 10% increase only....

Which will fail the above algorithm if someone if going above 240Tb.....

And the max supported we can go to is 768Tb

Reply:
(768-744)/744 = 3.2% approx

So we can have a limit saying atleast (threshod-3%) is full??

Reply:
The OSD filled percentage will always have some variability. 3% is too small. As discussed offline, horizontal scaling a small percentage such as 3% is really the issue. I'd suggest:
1. Use `threshold - 10%` for this secondary decision point (in this PR)
2. Horizontal scaling should increase the storage capacity at least 10% (in a separate PR). Instead of adding 3 OSDs, we will need to add some multiple of 3 OSDs to make it expand at least 10%. This means larger clusters will expand more proportionally.

Reply:
We can open a separate discussion of second point.

Reply:
@parth-gr @travisn are these new calculations affecting the way the UI calculates all the discrete values for the cluster capacity?

Reply:
@alfonsomthd I expect those discrete values will all still be valid. 
@parth-gr One consideration is that when we hit the max capacity, we only scale to that capacity (and not horizontally scale too many OSDs past that). Is that already in the calculation?

Reply:
If we are scaling horizontally, we add 3 osds, and if we cant add 3 more osd we do not scale, https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storageautoscaler/storageautoscaler_controller.go#L165

>I expect those discrete values will all still be valid.

It will change as we will be scaling to specific numbers

Reply:
> If we are scaling horizontally, we add 3 osds, and if we cant add 3 more osd we do not scale,

Understood, though let me give an example just to be clear... Let's say we have a cluster with max capacity 400TiB
and current capacity is 360TiB...
When we reach the 70% threshold:
   - Expansion of 3 OSDs (8TiB each) would take us to 384TiB capacity. 
   - This is only a 6.7% expansion and we want a minimum of 10% expansion if possible. 
   - So with the new proposal we would consider adding 6 OSDs to get at least 10% expansion. But that would give us 408TiB, which is over the 400TiB capacity, which is too much. 
   - So we just fall back to 3 OSDs expansion in this case.

Reply:
Agree, we can do that, but that also means, 10% is the min. increase?

as we have 3 osd with 8 tb and we add three more, we would be increasing 50% of the cluster.

Reply:
Yes, let's make 10% the min increase. For large clusters, horizontal scale of 3 OSDs is not enough of a jump.

Reply:
@parth-gr Can you update in the design doc the new algorithm? We need to update the calculations also on UI.

Reply:
@alfonsomthd as planned we will be doing it as part of 4.20